MHVLib/Clients/MHVRemoteMonitoringClient.m

```diff
@@ -104,4 +102,53 @@ - (void)requestWithPath:(NSString *_Nonnull)path
      }];
 }
 
+- (void)requestWithPath:(NSString *)path
+             httpMethod:(NSString *)httpMethod
+             pathParams:(NSDictionary<NSString *, NSString *> *_Nullable)pathParams
+            queryParams:(NSDictionary<NSString *, NSString *> *_Nullable)queryParams
+                   body:(NSData *_Nullable)body
+             completion:(void(^_Nullable)(NSError *_Nullable error))completion;
+{
+    MHVASSERT_PARAMETER(path);
+    MHVASSERT_PARAMETER(httpMethod);
+    
+    if (pathParams != nil)
+    {
+        NSMutableString *queryPath = [path mutableCopy];
+        
+        [pathParams enumerateKeysAndObjectsUsingBlock:^(id _Nonnull key, id _Nonnull obj, BOOL *_Nonnull stop)
+         {
+             [queryPath replaceCharactersInRange:[queryPath rangeOfString:[NSString stringWithFormat:@"{%@}", key]] withString:obj];
+         }];
+        
+        path = queryPath;
+    }
```

Comment:
Both methods here do this, I'd make it into a updatePath:withPathParams: method they can share.